packages/kit/src/internal/esm.ts

```diff
@@ -40,3 +41,31 @@ export function tryImportModule<T = unknown> (id: string, opts?: ImportModuleOpt
     // intentionally empty as this is a `try-` function
   }
 }
+
+const warnings = new Set<string>()
+
+/**
+ * @deprecated Please use `importModule` instead.
+ */
+export function requireModule<T = unknown> (id: string, opts?: ImportModuleOptions) {
+  if (!warnings.has(id)) {
+    console.warn('[@nuxt/kit] `requireModule` is deprecated. Please use `importModule` instead.')
```

Comment:
Just as idea, we might use `new Error().stack` to show the source of the warning. Most likely end-users won't have a clue of where is the origin...